src/System.Text.Json.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -167,8 +177,9 @@ public override bool TrySetMember(SetMemberBinder binder, object value)
         public bool TryFormat(Span<byte> buffer, out int written, TextFormat format, EncodingData formattingData)
         {
             written = 0;
+            int consumed;
             int justWritten;
-            if(!formattingData.TextEncoder.TryEncode('{', buffer, out justWritten)) {
+            if(!formattingData.TextEncoder.TryEncode(OpenCurlySpan, buffer, out consumed, out justWritten)) {
```

Comment:
Just going off from Krzysztof's comment:
> but there are real cases where we need to encode single chars, and we should think how to do it efficiently without creating a performance trap 

But I agree that we can add it later when required.